best-practices/README.md

```diff
@@ -0,0 +1,75 @@
+
+- What does it do and why does it do it?
+- Where is it deployed?
+- How are changes deployed?
+- Can I run it locally and if so how?
+- How to test?
+- Oddities ( for instance homepage uses AWS SDK 2 instead of 3 because of PHP5 )
+- What configuration items are there | Is there an initial (one-time) setup?
+- Are any credentials needed?
+
+If your README can answer all these questions you likely have a great README!
+
+#### Examples of Real READMEs
+
+Below are a few examples of READMEs we think are stellar examples:
+
+[bu-infographics](https://github.com/bu-ist/bu-infographics/blob/develop/README.md)
+
+### Commented Out Code
+
+When making changes to a remote Git repository ( e.g. Github.com ), particularly `master`, or otherwise shared code don't persist commented out code.
+
+- [Don't commit commented out code](https://kentcdodds.com/blog/please-dont-commit-commented-out-code).
+- [Don't share commented out code](https://www.nayuki.io/page/dont-share-commented-out-code).
+
+Code you're `pushing` or otherwise sharing should not include commented out code. Doing so makes reading your code take longer for **no gain**.
+
+By all means comment out code locally while working out a change but when sharing it by publishing it to github.com it should only contain used code. If you wish to save commented out code remotely just in case do so in a branch explicitly marked for your personal use only. Remember to remove any remaning commented out code though before merging to `master` or other shared branches.
+
+### Branches are cheap
+
+Speaking of branches, in Git branches are cheap and easy to use. Almost every workflow at BU will benefit from taking advantage of branches. Many codebases at BU use [Gitflow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow) to varying degrees but nearly all codebases utilize branching to ensure all code in `master` is production ready.
+
+Typically, the `master` branch is considered a production quality branch, or may even be production! That means changes you make and push to `master` should follow all the principles laid out in this repo. Code destined for `master` should be well-formed, readable and contain no commented out code or other artifacts that are not being deployed to production.
+
+#### Use branches for your code changes
+
+Some codebases at BU are new to Git and some have been Gitting for years. Read through any Contributing documentation to see how that repo is structured. If there is not Contributing doc, look in the README. If still in doubt, ask someone but in almost all cases creating a new branch will be the least disruptive and safest. Once your branched code is ready, merging into `master` ( or `develop` if following a workflow similar to the [`coding-standards`](https://github.com/bu-ist/coding-standards/blob/master/CONTRIBUTING.md) or whichever branch that repo's workflow designates as the merging branch ).
```

Comment:
Can we add a reference to the excellent Atlassian Git tutorials on branching here?

https://www.atlassian.com/git/tutorials/what-is-version-control

Reply:
Is that the right link? More of a general discussion of VC than branching specifically. Although maybe good to add this link as well?

Reply:
Put that link at the start: 

https://github.com/bu-ist/coding-standards/pull/44/commits/ce3bbed548a5dff149eb26c8d3ed7ba36c9cbda5

Reply:
I didn't see a better opening page to use for the whole series, so I use that as a beginning page.  Ultimately we might link to the site multiple times, once here and maybe again with branching.  Just having it in there seems like a good start though.